swift.en.md

```diff
@@ -0,0 +1,340 @@
+    }
+}
+```
+
+- **[MUST]** Do not use `@escaping` without needed
+- **[MUST]** Use `weak` instead of `unowned` as capture controls.
+    - In some cases, `unowned` improves performance. In other hand readability will be lost.
+
+## Property, Variable
+
+- **[MUST]** Use `let` to make variables immutable as possible.
+- **[MUST]** Drop type definition as possible. However, in the case of avoiding type inference for improvement compilation time, you can define clearly.
+
+### Property
+
+- **[MUST]** Use closest access control as possible.
+- **[MUST]** Use `private(set)` for the property want to write from limited scopes.
+- **[MUST]** Use `weak` as property attributes as possible.
+    - You should use `weak` to avoid circular references (eg. `@IBOutlet`, delegate)
+- **[MUST]** Do not use `unowned` as property attribute.
+    - In some cases, `unowned` improves performance. In other hand readability will be lost.
+
+```swift
+// Bad
+var delegate: FooDelegate
+@IBOutlet var textLabel: UILabel!
+
+// Good
+weak var delegate: FooDelegate
+@IBOutlet weak var textLabel: UILabel!
+```
+
+- **[MUST]** Do not use `get` block as possible on computed properties.
+
+```swift
+// Bad
+var foodCount: Int {
+    get { return foods.count }
+}
+
+// Good
+var foodCount: Int { 
+    return foods.count 
+}
+```
+
+- **[SHOULD]** If setting a property has a side-effect, write it in `willSet` or `didSet`.
+
+### Variable
+
+- **[MUST]** Do not use global variables.
+
+
+## Type
+
+- **[MUST]** Do not use `Array<T>`, `Dictionary<K, V>`. Use syntax sugar instead.
+
+```swift
+// Bad
+let seasonings: Array<String> = ["sugar", "salt", "vinegar", "soy sauce", "miso"]
+let ingredients: Dictionary<String, String> = [
+    "bacon": "100g",
+    "lettuce": "1/4",
+    "tomato": "half",
+]
+
+// Good let seasonings: [String] = ["sugar", "salt", "vinegar", "soy sauce", "miso"] let ingredients: [String: String] = [ "bacon": "100g",
+    "lettuce": "1/4",
+    "tomato": "half",
+]
+```
+
+- **[SHOULD]** Use `typealias` to represent complex type definitions.
+
+```swift
+// Good
+typealias RecipeClientCompletionBlock = (Result<[Recipe], APIError>) -> Void
+```
+
+- **[SHOULD]** Use classes without `NS` prefixes as possible.
+
+```swift
+// Bad
+let url = NSURL(string: "https://cookpad.com/")
+
+// Good
+let url = URL(string: "https://cookpad.com/")
+```
+
+
+- **[SHOULD]** Do not use `UInt`
+    - [Apple announced to avoid using it.](https://developer.apple.com/library/ios/documentation/Swift/Conceptual/Swift_Programming_Language/TheBasics.html#//apple_ref/doc/uid/TP40014097-CH5-ID320)
+
+## Enum
+
+- **[MUST]** Use lowerCamelCase as value naming.
+- **[MUST]** Drop enum type if you can.
+
+```swift
+// Bad
+user.status = UserStatus.guest
+
+// Good
+user.status = .guest
+```
+
+## Optional
+
+- **[MUST]** Do not use `Optional<T>`. Use syntax sugar instead.
+- **[MUST]** Use non-Optional value as possible.
+- **[MUST]** Do not use `guard` without optional-binding.
```

Comment:
Not fully understand this line. I don't think code with `guard` would compile with non-optional value.